### PR TITLE
Set terrain from diff

### LINKED ISSUE #3463: When updating the style with `diff:true` and terrain change the map doesn't change
There is a way to have the map look for changes in the style and apply them automatically, which is a great feature, but it doesn't take the terrain part of the spec into consideration.

Here's an example:
https://jsbin.com/muriqipaba/edit?html,output
When switching from `diff: true` to `diff:false` the 3D is removed, but it should also work for diff mode of update style.

Motivation:
I tried to add the terrain configuration to maputnik only to discover this doesn't work due to this issue.
Also maputnik doesn't use typescript, so upgrading to version 3/4 from the current version (2.4) will be a nightmare...

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist

Fixes #3463 - add support for terrain in diff
- #3463

This depends on the following PR to be merged and published first:
-  https://github.com/maplibre/maplibre-style-spec/pull/460

Also added some types to `pick` method.

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
 - [x] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
The 3 missing lines in the coverage report are related to a situation where the diffStyle function from the style spec package returns a different operation than expected.
I'm not sure it's worth the extra effort of trying to mock the diffStyle method in order to cover this code, it's mostly a safeguard.

This is now ready for review.

File: src/util/util.test.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ describe('util', () => {
     expect(keysDifference({a: 1}, {a: 1})).toEqual([]);
     expect(extend({a: 1}, {b: 2})).toEqual({a: 1, b: 2});
     expect(pick({a: 1, b: 2, c: 3}, ['a', 'c'])).toEqual({a: 1, c: 3});
-    expect(pick({a: 1, b: 2, c: 3}, ['a', 'c', 'd'])).toEqual({a: 1, c: 3});
+    expect(pick({a: 1, b: 2, c: 3}, ['a', 'c', 'd'] as any)).toEqual({a: 1, c: 3});
```
Comment (birkskyum) [inline src/util/util.test.ts:14]: why is `as any` needed here, but not on the line above?

Comment (HarelM) [reply src/util/util.test.ts:14]: because `d` is not part of `{a, b, c}` object.

Comment (birkskyum) [reply src/util/util.test.ts:14]: I was wondering if it's better to change the KeysOfUnion code to allow for attempting to pick non-existing keys, but i guess that would just remove useful mis-match information.

**Review by birkskyum — APPROVED** [review]

**Review by neodescis — APPROVED** [review]

**Pull request merged**